### DISCUSSION TITLE
Use of spaCy to compute Syntactic Dependency Length

### DISCUSSION THREAD

**Discussion by Arka-cell** [explosion/spaCy#6723] [Show & Tell]
Hi, I have found out that spaCy proved to be very accurate at computing dependency lengths for language and I used it during my master thesis in Language Studies. I have created a repo that you can find [here](https://github.com/Arka-cell/DepLength), if you have any ideas on how to improve the code or give more formulas to calculate dependency lengths (I have made the most basic formula) just commit your changes. Research about dependency lengths in language studies is common but I found that no one really made a step to make it accessible, what blew my mind is how easy and straightforward it was to make it possible with spaCy. 

For further information on the algorithm used see the following thread in [math.stackexchange](https://math.stackexchange.com/questions/3775965/what-is-the-mathematical-notation-for-dependency-length-calculation-algorithm). Big thanks to gfppoy for the incredible explanation for the algorithm. Also, you can directly pip install it from [here](https://pypi.org/project/depLength/).

**Comment by svlandeg** [comment]
Oh, cool, thanks for sharing!
Perhaps your readme file could show a few very simple examples, showcasing the API and the type of results you get back?

**Reply by Arka-cell** [reply]
Hey Sofia, thanks! 
I noticed that there were some errors in the pip version, it should be okay now. I have added an .html page that explains what happens alongside a demo code that can show how to get the result of a dependency length. So, it should be good to to go from here.  

After you <code>pip install deplength</code> you can try the following code. You might also want to read Formula.pdf to see the underlying concept, you can now find it in [here](https://github.com/Arka-cell/DepLength/blob/main/Formula.pdf). 

Thanks for your interest!

<code>
from depLength import depLength
</code>
<br>
<code>sentences = """
John threw out the trash. John threw the trash out. John threw out the old trash sitting in the kitchen. John threw the old trash sitting in the kitchen out.
"""</code>

res = depLength.DepLength(sentences)
print(res.sdl())
</code>
